### PR TITLE
Support webpack 4.0 tapable api

### PR DESCRIPTION
**Pull request by ccorcos**
Fix the deprecation warning for webpack 4.0 https://github.com/lettertwo/appcache-webpack-plugin/issues/23

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -69,13 +69,12 @@ export default class AppCachePlugin {
     const {options: {output: outputOptions = {}} = {}} = compiler;
     const {publicPath = ''} = outputOptions;
 
-    compiler.plugin('emit', (compilation, callback) => {
+		compiler.hooks.emit.tap("appacache-webpack-plugin", (compilation) => {
```
Comment (lettertwo) [inline src/index.js]: Let's change the plugin name to `'AppCachePlugin'`.

File: src/index.js [hunk 2]
```diff
@@ -69,13 +69,12 @@ export default class AppCachePlugin {
     const {options: {output: outputOptions = {}} = {}} = compiler;
     const {publicPath = ''} = outputOptions;
 
-    compiler.plugin('emit', (compilation, callback) => {
```
Comment (lettertwo) [inline src/index.js:72]: I'd like to maintain backwards compatibility here, if possible. For instance, we could detect whether `compiler.hooks.emit.tap` is available, but fall back to `compiler.plugin` if it isn't.

**Review by lettertwo — CHANGES_REQUESTED** [review]
Hi @ccorcos, thanks for the PR!

I made a couple of notes about changes i'd like before merging this. If you have the time, I would greatly appreciate it!

**Comment by ccorcos** [comment]
hey @lettertwo I addressed your comments

**Comment by ccorcos** [comment]
ping @lettertwo

**Comment by briandipalma** [comment]
Is this OK to merge?

**Comment by lettertwo** [comment]
Sorry for the delay, everyone. I've merged this PR, but some tests are failing, and there are some linting nits i'm going to clean up before i cut a release.

Thanks for the PR @ccorcos!

**Comment by lettertwo** [comment]
This is now published as [v1.4.0](https://www.npmjs.com/package/appcache-webpack-plugin/v/1.4.0). Thanks again for your PR and patience!

**Comment by briandipalma** [comment]
Cheers, thank you.

**Pull request merged**